File: app/src/main/java/com/sanyuan/floodprevention09/activity/PillarProvinceActivity.java
package com.sanyuan.floodprevention09.activity;

import android.os.Build;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.annotation.RequiresApi;
import android.transition.ChangeBounds;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.sanyuan.floodprevention09.R;
import com.sanyuan.floodprevention09.base.BaseActivity;
import com.sanyuan.floodprevention09.customview.PillarViewProvince;

/**
 * @author wodemingziyouyidianchang
 * @version 1.0.0
 * @since 2018/6/27 17:41
 */
public class PillarProvinceActivity extends BaseActivity implements View.OnClickListener {

    private PillarViewProvince bcv;
    private ImageView iv_back;
    private TextView tv_title;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

//        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
//            share();
//        }
    }

    @Override
    protected int getMyViewId() {
        return R.layout.act_pillarprovince;
    }

    @Override
    protected void initView() {
        bcv = (PillarViewProvince) findViewById(R.id.bcv);

//        tv_title = (TextView) findViewById(R.id.tv_title);
//        tv_title.setText("全省易涝点");

        iv_back = (ImageView) findViewById(R.id.iv_back);
        iv_back.setOnClickListener(this);
        iv_back.setVisibility(View.VISIBLE);

    }

    @Override
    protected void initData() {

    }

    @Override
    public void onClick(View view) {
        int id = view.getId();

        switch (id) {

            case R.id.iv_back:
                finish();
                break;

        }

    }


    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
    private void share() {

        ChangeBounds cb = new ChangeBounds();
        cb.addTarget("share");
        cb.setDuration(2000);

//        Toast.makeText(getApplicationContext(), "共", Toast.LENGTH_SHORT).show();

        getWindow().setSharedElementEnterTransition(cb);
        getWindow().setSharedElementExitTransition(cb);


    }

}

File: app/src/main/java/com/sanyuan/floodprevention09/activity/PillarProvinceActivity.kt
package com.sanyuan.floodprevention09.activity

import android.os.Build
import android.os.Bundle
import android.support.annotation.RequiresApi
import android.transition.ChangeBounds
import android.view.View
import android.widget.ImageView
import android.widget.TextView

import com.sanyuan.floodprevention09.R
import com.sanyuan.floodprevention09.base.BaseActivity
import com.sanyuan.floodprevention09.customview.PillarViewProvince

/**
 * @author wodemingziyouyidianchang
 * @version 1.0.0
 * @since 2018/6/27 17:41
 */
class PillarProvinceActivity : BaseActivity(), View.OnClickListener {

    private var bcv: PillarViewProvince? = null
    private var iv_back: ImageView? = null
    private val tv_title: TextView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        //        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
        //            share();
        //        }
    }

    override fun getMyViewId(): Int {
        return R.layout.act_pillarprovince
    }

    override fun initView() {
        bcv = findViewById(R.id.bcv) as PillarViewProvince

        //        tv_title = (TextView) findViewById(R.id.tv_title);
        //        tv_title.setText("全省易涝点");

        iv_back = findViewById(R.id.iv_back) as ImageView
        iv_back!!.setOnClickListener(this)
        iv_back!!.visibility = View.VISIBLE

    }

    override fun initData() {

    }

    override fun onClick(view: View) {
        val id = view.id

        when (id) {

            R.id.iv_back -> finish()
        }

    }


    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
    private fun share() {

        val cb = ChangeBounds()
        cb.addTarget("share")
        cb.duration = 2000

        //        Toast.makeText(getApplicationContext(), "共", Toast.LENGTH_SHORT).show();

        window.sharedElementEnterTransition = cb
        window.sharedElementExitTransition = cb


    }

}
